docs/excel/excel-data-types-linked-entity-cell-values.md

```diff
@@ -0,0 +1,539 @@
+- `LinkedEntityDataDomainCollection.refreshAll()` - Refreshes all `LinkedEntityCellValue` objects of all linked entity data domains in the collection.
+
+The refresh methods request a refresh which occurs asynchronously. To determine the results of the refresh, listen for the `onRefreshCompleted` event. The following code sample shows an example of listening for the `onRefreshCompleted` event.
+
+```javascript
+ await Excel.run(async (context) => {
+    const dataDomains = context.workbook.linkedEntityDataDomains;
+    dataDomains.onRefreshCompleted.add(onLinkedEntityDomainRefreshed);
+
+    await context.sync();
+});
+
+async function onLinkedEntityDomainRefreshed (eventArgs: Excel.LinkedEntityDataDomainRefreshCompletedEventArgs): Promise<any> {
+  console.log("Linked entity domain refreshed: " + eventArgs.id);
+  console.log("Refresh status: " + eventArgs.refreshed);
+  console.log("Refresh error: " + eventArgs.errors);
+  return null;
+}
+```
+
+## Error handling in linked entity load service
+
+When Excel calls your add-in to get data for a linked entity cell value, it's possible an error can occur. If Excel is unable to connect to your add-in at all (such as when it is not loaded) it will display `#CONNECT!` to the user.
+
+If your linked entity load service function encounters an error, it should throw an error. The error will cause Excel to show `#CONNECT!` to the user.
+
+The following code shows how to handle an error in a linked entity load service function.
+
+```javascript
+async function productLinkedEntityService(request: any): Promise<any> {
+    const notAvailableError = new CustomFunctions.Error(CustomFunctions.ErrorCode.notAvailable);
+    try {
+      // Create an return a new linked entity cell value.
+        let linkedEntityResult = ...
+      ...
+        if (!linkedEntityResult) {
+        // Throw an error to signify to Excel that resolution/refresh of the requested linkedEntityId failed.
+        throw notAvailableError;
+      }
+      ...
+    } catch (error) {
+        console.error(error);
+        throw notAvailableError;
+    }
+}
+```
+
+## Debugging the linked entity load service
+
+Most add-in functionality for linked entity data types can be debugged using the guidance in [Overview of debugging Office Add-ins](../testing/debug-add-ins-overview.md). However, the linked entity load service function can be implemented in a shared runtime, or a JavaScript-only runtime (also know as a custom function runtime.) If you choose to implement the function in a JavaScript-only runtime, use the debugging guidance in JavaScript-only runtime for [Custom functions debugging in a non-shared runtime](custom-functions-debugging.md).
+
+The linked entity load service function uses the custom function architecture, regardless of which runtime you use. However, there are significant differences from regular custom functions.
+Differences:
+
+- They will not appear to end users for usage in formulas.
```

Comment:
What is "they" here? Linked entities? Is it specifically "Linked entity load service functions"?

Reply:
reworked the intros to be clearer.